File: VB-Examples/07_Conversion/WorkbookToHTML.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace WorkbookToHTML
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of the Workbook class.
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\WorkbookToHTML.xlsx")

            ' Save the workbook to an HTML file.
            workbook.SaveToHtml("result.html")
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer("result.html")
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/07_Conversion/WorkbookToHTML.cs
using Spire.Xls;
using System;
using System.Windows.Forms;

namespace WorkbookToHTML
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load a file from the specified path into the workbook
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\WorkbookToHTML.xlsx");

            //Convert to html
            workbook.SaveToHtml("result.html");

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer("result.html");
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
